app/com/linkedin/drelephant/spark/heuristics/StagesAnalyzer.scala

```diff
@@ -0,0 +1,498 @@
+          "the partitions more even."
+      }
+    }
+    val score = Utils.getHeuristicScore(taskSkewSeverity, stageData.numTasks)
+
+    TaskSkewResult(taskSkewSeverity, score, details, rawSkewSeverity)
+  }
+
+  /**
+    * Check for skewed data.
+    *
+    * @param stageId stage ID
+    * @param median median data size for tasks.
+    * @param maximum maximum data size for tasks.
+    * @param description type of data.
+    * @param details information and recommendations -- any new recommendations
+    *                from analyzing the stage for data skew will be appended.
+    */
+  private def checkSkewedData(
+      stageId: Int,
+      median: Double,
+      maximum: Double,
+      description: String,
+      details: ArrayBuffer[String]): Severity = {
+    val severity = taskSkewThresholds.severityOf(maximum / median)
+    if (hasSignificantSeverity(severity)) {
+      details += s"Stage $stageId has skew in $description (median is " +
+        s"${MemoryFormatUtils.bytesToString(median.toLong)}, " +
+        s"max is ${MemoryFormatUtils.bytesToString(maximum.toLong)})."
+    }
+    severity
+  }
+
+  /**
+    * Check the stage for long running tasks.
+    *
+    * @param stageId stage ID.
+    * @param stageData stage data.
+    * @param medianTime median task run time.
+    * @param curNumPartitions number of partitions for the Spark application
+    *                         (spark.sql.shuffle.partitions).
+    * @return results of long running task analysis for the stage
+    */
+  private def checkForLongTasks(
+      stageId: Int,
+      stageData: StageData,
+      medianTime: Option[Double],
+      curNumPartitions: Int): SimpleStageAnalysisResult = {
+    val longTaskSeverity = stageData.taskSummary.map { distributions =>
+      taskDurationThresholds.severityOf(distributions.executorRunTime(DISTRIBUTION_MEDIAN_IDX))
+    }.getOrElse(Severity.NONE)
+    val details = new ArrayBuffer[String]
+    if (hasSignificantSeverity(longTaskSeverity)) {
+      val runTime = Utils.getDuration(medianTime.map(_.toLong).getOrElse(0L))
+      val maxData = Seq(stageData.inputBytes, stageData.shuffleReadBytes, stageData.shuffleWriteBytes,
+        stageData.outputBytes).max
+      details += s"Stage $stageId: median task run time is $runTime."
+      if (stageData.numTasks >= maxRecommendedPartitions) {
+        if (maxData >= maxDataProcessedThreshold) {
+          val inputBytes = MemoryFormatUtils.bytesToString(stageData.inputBytes)
+          val shuffleReadBytes = MemoryFormatUtils.bytesToString(stageData.shuffleReadBytes)
+          val shuffleWriteBytes = MemoryFormatUtils.bytesToString(stageData.shuffleWriteBytes)
+          details += s"Stage $stageId has $inputBytes input, $shuffleReadBytes shuffle read, " +
+            "$shuffleWriteBytes shuffle write. Please try to reduce the amount of data being processed."
+        } else {
+          details += s"Stage $stageId: please optimize the code to improve performance."
```

Comment:
Where are we planning to use this details . How are we planning to show this to user ? . Is it going to be used in Heuristics ?

Reply:
Yes, the current plan is that the details would be shown in Heuristics. This could be pretty messy if there are a lot of lines, however. Is there a better way to present the information?

Reply:
Yes showing details on heuristics , would make UI pretty messy .   I think we can add , detail button on each heuristics , clicking that can show this detail  (again I am not sure  other ways to represent details)

Reply:
It would be a bit more hidden, but a button that would show the details would be cleaner.